R/sample_predictions.R

```diff
@@ -107,22 +128,19 @@ sample_prediction <- function(point_nowcast_matrix,
 #' )
 #' reporting_triangle <- construct_triangle(reporting_matrix)
 #' combine_obs_with_pred(pred_counts, reporting_triangle)
-#'
 #' # Another example with rolling sum
-#' combine_obs_with_pred(pred_counts, reporting_triangle, k = 2)
-combine_obs_with_pred <- function(predicted_counts,
-                                  reporting_triangle,
-                                  fun_to_aggregate = sum,
-                                  k = 1) {
-  .validate_aggregation_function(fun_to_aggregate)
-  obs_counts <- rollapply(
-    rowSums(reporting_triangle, na.rm = TRUE),
-    k,
-    fun_to_aggregate,
-    fill = NA,
-    align = "right"
-  )
-  return(obs_counts + predicted_counts)
+#' combine_obs_with_pred(pred_counts,
+#'   reporting_triangle,
+#'   ref_time_aggregator = function(x) zoo::rollsum(x, k = 2, align = "right")
+#' )
+combine_obs_with_pred <- function(
```

Comment:
just in general there is way too much functionality living in utils suggests issue for resort

Reply:
Have made a new issue https://github.com/epinowcast/baselinenowcast/issues/230